client/compare.go

```diff
@@ -0,0 +1,127 @@
+package client
+
+import xlineapi "github.com/xline-kv/go-xline/api/xline"
+
+type CompareTarget int
+type CompareResult int
+
+const (
+	CompareVersion CompareTarget = iota
+	CompareCreated
+	CompareModified
+	CompareValue
+)
+
+type Cmp xlineapi.Compare
+
+// nolint: govet
+func Compare(cmp Cmp, result string, v interface{}) Cmp {
+	var r xlineapi.Compare_CompareResult
+
+	switch result {
+	case "=":
+		r = xlineapi.Compare_EQUAL
+	case "!=":
+		r = xlineapi.Compare_NOT_EQUAL
+	case ">":
+		r = xlineapi.Compare_GREATER
+	case "<":
+		r = xlineapi.Compare_LESS
+	default:
+		panic("Unknown result op")
+	}
+
+	cmp.Result = r
+	switch cmp.Target {
+	case xlineapi.Compare_VALUE:
+		val, ok := v.(string)
+		if !ok {
+			panic("bad compare value")
+		}
+		cmp.TargetUnion = &xlineapi.Compare_Value{Value: []byte(val)}
+	case xlineapi.Compare_VERSION:
+		cmp.TargetUnion = &xlineapi.Compare_Version{Version: mustInt64(v)}
+	case xlineapi.Compare_CREATE:
+		cmp.TargetUnion = &xlineapi.Compare_CreateRevision{CreateRevision: mustInt64(v)}
+	case xlineapi.Compare_MOD:
+		cmp.TargetUnion = &xlineapi.Compare_ModRevision{ModRevision: mustInt64(v)}
+	case xlineapi.Compare_LEASE:
+		cmp.TargetUnion = &xlineapi.Compare_Lease{Lease: mustInt64orLeaseID(v)}
+	default:
+		panic("Unknown compare type")
+	}
+	return cmp
+}
+
+func Value(key string) Cmp {
+	return Cmp{Key: []byte(key), Target: xlineapi.Compare_VALUE}
+}
+
+func Version(key string) Cmp {
+	return Cmp{Key: []byte(key), Target: xlineapi.Compare_VERSION}
+}
+
+func CreateRevision(key string) Cmp {
+	return Cmp{Key: []byte(key), Target: xlineapi.Compare_CREATE}
+}
+
+func ModRevision(key string) Cmp {
+	return Cmp{Key: []byte(key), Target: xlineapi.Compare_MOD}
+}
+
+// LeaseValue compares a key's LeaseID to a value of your choosing. The empty
+// LeaseID is 0, otherwise known as `NoLease`.
+func LeaseValue(key string) Cmp {
+	return Cmp{Key: []byte(key), Target: xlineapi.Compare_LEASE}
+}
+
+// KeyBytes returns the byte slice holding with the comparison key.
+func (cmp *Cmp) KeyBytes() []byte { return cmp.Key }
+
+// WithKeyBytes sets the byte slice for the comparison key.
+func (cmp *Cmp) WithKeyBytes(key []byte) { cmp.Key = key }
+
+// ValueBytes returns the byte slice holding the comparison value, if any.
+func (cmp *Cmp) ValueBytes() []byte {
+	if tu, ok := cmp.TargetUnion.(*xlineapi.Compare_Value); ok {
+		return tu.Value
+	}
+	return nil
+}
+
+// WithValueBytes sets the byte slice for the comparison's value.
+func (cmp *Cmp) WithValueBytes(v []byte) { cmp.TargetUnion.(*xlineapi.Compare_Value).Value = v }
```

Comment:
Why isn't its function signature `func(cmp Cmp) WithValueBytes(v []byte) Cmp`?

Reply:
After we change the function parameter type to `[]byte`, we don't seem to need this method anymore.
FYI: https://github.com/xline-kv/go-xline/pull/15#discussion_r1377000652